### ISSUE TITLE
10x Performance Drop

### ISSUE THREAD

**Issue by Qendolin** [Traben-0/EmissiveMod#11]
I am getting a 13x performance drop when using this mod with a resource pack loaded.

## Environment

- 8 GB Memory, 4 Allocated to Minecraft
- GeForce 1650 with 4 GB VRAM
- Intel i7-9750H 2.6GHz@6 Cores/12 Threads
- Minecraft 1.18,1
- Mods: Entity Texture Features 2.1.2, Mod Menu 3.0.0, Fabric API 0.46.1

## Test Scene

- 120 cows
- Resource Pack: https://www.curseforge.com/minecraft/texture-packs/creature-variety/files/3493096
- Video Settings on lowest except 'Mipmap Levels', unlimited FPS

### Creature Variety enabled
![image](https://user-images.githubusercontent.com/32160662/152038376-23d0dd80-78cf-451f-8cf8-e5dcc349272c.png)
### Creature Variety disabled
![image](https://user-images.githubusercontent.com/32160662/152038484-ebbfaf15-4ccb-4704-b642-ba775a7362b4.png)

## Log
https://paste.ee/p/QHUdm
The log is showing a ton of errors, this *can* be the cause of the performance problem.

```
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "java.util.Map.get(Object)" is null
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: counting optifine properties failed
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: counting optifine properties failed
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: counting optifine properties failed
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: Entity Texture Features - optifine properties failed to load: optifine/random/entity/cow/cow.properties
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "java.util.Map.get(Object)" is null
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: counting optifine properties failed
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: counting optifine properties failed
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: counting optifine properties failed
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: Entity Texture Features - optifine properties failed to load: optifine/random/entity/cow/cow.properties
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "java.util.Map.get(Object)" is null
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: counting optifine properties failed
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: counting optifine properties failed
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: counting optifine properties failed
[20:30:09] [Render thread/INFO]: [Entity Texture Features]: Entity Texture Features - optifine properties failed to load: optifine/random/entity/cow/cow.properties
[...]
```

**Comment by Qendolin** [comment]
Here's my [test world](https://github.com/Traben-0/EmissiveMod/files/7981525/Entity.Textures.zip)

**Comment by Traben-0** [comment]
I have it working it was a problem with the file reading on my part every pack maker has different quirks that i may not expect
Fix will be in V2.1.3 sometime later today

BTW just a heads up but it doesn't look like the custom models in the pack work with CEM tried it both with and without my mod.
so the textures don't wrap around the vanilla model because its expecting a custom model, but they are randomising correctly now :/
![2022-02-02_14 39 36](https://user-images.githubusercontent.com/8456487/152094095-ed4daff8-0766-49e9-a83d-4f1509102490.png)
